R/utils.R

```diff
@@ -288,28 +288,47 @@ check_attributes <- function(attributes, call = caller_env()) {
 }
 
 # apple + spark drive config (#651) --------------------------------------------
-configure_spark <- function(call = caller_env()) {
+# Method will attempt to:
+# 1. Locate the simba driver config using the user supplied callback.  Callback
+# method defaults to locate_config_spark.
+# 2. Inspect the config for some settings that can impact how our package
+# performs.
+# 3. If action == "modify" then we attempt to modify the config in-situ.
+# 4. Otherwise we throw a warning asking the user to revise.
+configure_simba <- function(locate_config_callback = (function() character()),
+                            action = "modify", call = caller_env()) {
   if (!is_macos()) {
     return(invisible())
   }
-
+  if (is.null(getOption("odbc.no_config_override"))) {
+    return(invisible())
+  }
   unixodbc_install <- locate_install_unixodbc()
   if (length(unixodbc_install) == 0) {
     error_install_unixodbc(call)
   }
 
-  spark_config <- locate_config_spark()
-  if (length(spark_config) == 0) {
-    abort(
+  simba_config <- locate_config_callback()
+  if (length(simba_config) == 0) {
+    func <- cli::warn
+    if ( action == "modify" ){
+      fun <- cli::abort
+    }
+    func(
       c(
-        "Unable to locate the needed spark ODBC driver.",
-        i = "Please install the needed driver from https://www.databricks.com/spark/odbc-drivers-download."
+        "Unable to locate the config for the ODBC driver.",
+        i = paste("Please make sure you've installed the appropriate driver",
+          "for your platform and back end."),
```

Comment:
```suggestion
        i = "Please make sure you've installed the appropriate driver
             for your platform and back end."),
```

Reply:
GitHub doesn't seem to properly format these changes, but cli allows you to add newlines in error messages with no further formatting, i.e. no need to `paste()`. This can also apply to the next two lines, but I think these three `i` elements could likely be collapsed into a single:

```
i = "Please install the needed driver from {substitute_in_the_appropriate_link_based_on_dbms}"
```

Reply:
Done / thanks.